### PR TITLE
Detonate Drones with RD console

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: You can now detonate drones with the RD console
/:cl:

Cogscarabs, FREEDRONES, and syndrones are all immune

### CODE REVIEW

**Comment by GunHog** [comment]
Why would anyone want to blow drones? Also, good luck telling them apart with just their number.

**Comment by lzimann** [comment]
because drones are annoying, and you can just blow them all to be safe

**Comment by KorPhaeron** [comment]
@GunHog 

When you're antag and a swarm of butthurt ghost drones are "accidentally" interfering with you by fixing windows you just stepped through, fighting fires you lit, grabbing random shit off the floor, and you're unable to kill the tiny bastards because they're non dense.

The admins are often lost and confused in such situations as well because they all look similar and have unmemorable names.

**Comment by Supermichael777** [comment]
drones are easy to kill but the metal tide wont stop self repliccating

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm [hunk 1]
```diff
@@ -110,6 +111,7 @@
 	laws = "0. Purge all untruths and honor Ratvar."
 	default_storage = /obj/item/weapon/storage/toolbox/brass/prefilled
 	seeStatic = 0
+	hacked = TRUE
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: line 104

File: code/game/machinery/computer/robot.dm [hunk 2]
```diff
@@ -68,6 +68,21 @@
 
 	if(!robots)
 		dat += "No Cyborg Units detected within access parameters."
+		dat += "<BR>"
+
+	var/drones = 0
```
Comment (ChangelingRain) [inline code/game/machinery/computer/robot.dm]: You never increase this var

File: code/game/machinery/computer/robot.dm [hunk 3]
```diff
@@ -68,6 +68,21 @@
...
 		dat += "No Cyborg Units detected within access parameters."
+		dat += "<BR>"
+
+	var/drones = 0
+	for(var/mob/living/simple_animal/drone/D in mob_list)
+		if(D.hacked)
+			continue
+		dat += "[D.name] |"
+		if(D.stat)
+			dat += " Not Responding |"
```
Comment (ChangelingRain) [inline code/game/machinery/computer/robot.dm]: Shouldn't have the `|` at the end because nothing follows it

**Comment by TechnoAlchemisto** [comment]
the drone dispenser in RnD was a bad addition

**Comment by GunHog** [comment]
So, policy wise, when may a player use this against drones, and not be punished as per Rule 1? Does he need to be an antagonist? May he blow drones whom have stolen things from him? What if he tries to get a drone to open a door for him, and it ignores him? What about if the player just thinks it is funny to gib drones?

**Comment by PKPenguin321** [comment]
pretty sure you can just kill drones for whatever anyways so it doesnt really matter
drones are a highly disposable ghost role

**Comment by optimumtact** [comment]
I agree with @TechnoAlchemisto 

Coiax buffed drones mercilessly when he was still around.

**Comment by Core0verload** [comment]
I can't :+1: this enough. Burn in hell, shit drones.

**Comment by Militaires** [comment]
the only bad part is that shitfucks who kill drones for shits and giggles can just waltz up to a console barabeem baraboom.

**Comment by Alek2ander** [comment]
Missing a "DESTROY ALL DRONES" button

**Comment by Kromgar** [comment]
I'd hit the destroy all drones button every 5 minutes

**Comment by Durkel** [comment]
I can't help but feel I'm somewhat responsible for this.

**Pull request merged**